src/components/Checkbox/Checkbox.jsx

```diff
@@ -0,0 +1,116 @@
+import PropTypes from 'prop-types'
+import React, { forwardRef } from 'react'
+import styled, { css, down, up, typography } from '@xstyled/styled-components'
+
+import { Typography } from '../'
+import { Icon } from '../Iconography'
+
+const Checkbox = forwardRef(({ color, label, ...props }, ref) => {
+  return (
+    <LabelContainer color={color}>
+      <Input type='checkbox' ref={ref} {...props} />
+      <Checkmark />
+      {label && (
+        <Typography fontSize={3} lineHeight={3} fontWeight={0} marginLeft={6}>
+          {label}
+        </Typography>
+      )}
+    </LabelContainer>
+  )
+})
+
+Checkbox.propTypes = {
+  checked: PropTypes.bool,
+  color: PropTypes.string,
+  disabled: PropTypes.bool,
+  onChange: PropTypes.func,
+  label: PropTypes.string,
+  name: PropTypes.string
+}
+
+const LabelContainer = styled.label`
+  display: inline-flex;
+  vertical-align: top;
+  position: relative;
+  user-select: none;
+  margin-left: 0;
+  color: gray.800;
+  min-height: 24px;
+  min-width: 24px;
+
+  ${down(
+    'sm',
+    css`
+      display: flex;
+    `
+  )}
+
+  &
+    ~ input:checked
+    p {
+    color: gray.900;
+  }
+  ${typography}
+`
+
+const Input = styled.input`
+  position: absolute;
+  opacity: 0;
+  height: 100%;
+  width: 100%;
+  margin: 0;
+  cursor: pointer;
+  z-index: 1;
```

Comment:
a ideia desse input é ficar completamente invisível. acho que remover todo esse bloco de css (mantendo apenas os seletores compostos) e no input colocar um `display: none`. Fazendo isso, podes passar a responsabilidade do controle de cursor (pointer e not-allowed) para os filhos diretos que são o checkmark e o texto.

```suggestion
  display: none;
```

Reply:
a ideia foi justamente não mostrar o Input, colocando a sua opacidade em 0 e ele estando acima de todos os outros com o Z-index. Mas por que isso? dessa forma eu preciso controlar os cursores apenas no Input, o que pra mim fez sentido, sem isso, eu tenho de controlar em outros dois lugares, que são o checkmark e a label, se o estilo do componente for mudado posteriormente (exemplo: alteração de padding e margins) eu não preciso me preocupar com isso, porque sempre o Input vai estar por cima de todos os outros elementos e vai controlar o cursor

Reply:
a ideia é evitar os positions absolutes ao máximo. sobrepor componentes não é o ideal pq estamos pensando em um projeto que não sabemos onde os componentes serão usados e nem como serão usados. esconder o Input 100% e controlar o cursor em cada um dos filhos é ideal pra garantir que mesmo em outros projetos essas "inconsistências" entre a forma que cada browser renderiza o CSS não impacte em um elemento com position absolute que está sobreposto a outros elementos.